phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -421,7 +514,20 @@ private PhoenixResultSet executeQuery(final CompilableStatement stmt,
                                     }
                                 }
                                 throw e;
-                            } catch (RuntimeException e) {
+                            } catch (SQLException e) {
+                                // force update cache if StaleMetadataCacheException and retry
+                                if (e instanceof StaleMetadataCacheException) {
+                                    String planSchemaName = getLastQueryPlan().getTableRef().getTable().getSchemaName().toString();
+                                    String planTableName = getLastQueryPlan().getTableRef().getTable().getTableName().toString();
+                                    // update cache
+                                    new MetaDataClient(connection).updateCache(connection.getTenantId(), planSchemaName, planTableName, true);
+                                    // skip last ddl timestamp validation in the retry
+                                    setValidateLastDdlTimestamp(false);
```

Comment:
Yeah I was thinking of that, do we ever re-use the statement for different queries? I could only find usage of `conn.createStatement` in our tests but we can not guarantee that usage from customers. 

I can revert back to passing a boolean parameter through the execute* methods.

Reply:
You can use the same prepared statement object in a loop by binding different values to the parameters in the query and then calling execute query on it.

Reply:
Just keeping a class variable to store the config property, it can be used every time for executeQuery or executeMutation even if the same Statement object is used.